reverse_proxy_test.go

```diff
@@ -132,3 +132,137 @@ func TestRequestIP(t *testing.T) {
 		}
 	}
 }
+
+func TestCheckHeaderInRemoveList(t *testing.T) {
+	tests := []struct {
+		header   string
+		spec     string
+		expected bool
+	}{
+		{
+			header:   "X-Forwarded-For",
+			spec:     "",
+			expected: false,
+		},
+		{
+			header: "X-Forwarded-For",
+			spec: `{
+				"api_id": "1",
+				"version_data": {
+					"not_versioned": true,
+					"versions": {
+						"Default": {
+							"name": "Default",
+							"use_extended_paths": true,
+							"global_headers_remove": [ "X-Random-Header" ]
+						}
+					}
+				}
+			}`,
```

Comment:
You could simplify this test by just having the version definition in each of these strings, like:

```
version: `{
"name": "Default",
"use_extended_paths": true,
"global_headers_remove": [ "X-Random-Header" ]
 }
```

The rest is always the same, so you can just use a `text/template.Template` to put it together.